File: 03_Exception/ReadChar.c
#include <stdio.h>
#include <errno.h>

int main(void) {
    FILE *fp;
    char c, text[100000] = "";
    int i = 0;

    fp = fopen("ReadChar.java", "r");
    if (fp == NULL) {
        puts("An error occurred.");
        return 1;
    }

    while ((c = fgetc(fp)) != EOF) {
        text[i++] = c;
    }
    text[i] = '\0';     // 終端文字追加
    fclose(fp);

    puts(text);

    return 0;
}

File: 03_Exception/ReadChar.java
import java.io.*;

public class ReadChar {
    public static void main(String[] args) {
        int c;
        String text = "";

        try {
            FileReader in = new FileReader("ReadChar.java");

            while ((c = in.read()) != -1) {
                text += (char) c;
            }

            in.close();
        } catch (IOException e) {
            System.out.println("File not found.");
        } catch (Exception e) {
            System.out.println("An error occurred.");
        }

        System.out.println(text);
    }
}


